lib/hammer_cli_foreman/user_mail_notification.rb

```diff
@@ -0,0 +1,54 @@
+module HammerCLIForeman
+
+  class UserMailNotification < HammerCLIForeman::Command
+    resource :mail_notifications
+    command_name 'mail-notification'
+
+    class AddMailNotificationCommand < HammerCLIForeman::CreateCommand
+      action :create
+      command_name 'add'
+
+      success_message _('User mail notifications was added.')
+      failure_message _('Could not add user mail notification')
+
+      build_options
+    end
+
+    class UpdateMailNotificationCommand < HammerCLIForeman::UpdateCommand
+      action :update
+      command_name 'update'
+
+      success_message _('User mail notifications was updated.')
+      failure_message _('Could not update user mail notification')
+
+      build_options
+    end
+
+    class RemoveMailNotificationCommand < HammerCLIForeman::DeleteCommand
+      action :destroy
+      command_name 'remove'
+
+      success_message _('User mail notification was removed.')
+      failure_message _('Could not remove user mail notification')
+
+      build_options
+    end
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      action :user_mail_notifications
+
+      output do
+        field :mail_notification_id, _('Id')
+        field :name, _('Name')
+        field :description, _('Description')
+        field :interval, _('Interval')
+        field :mail_query, _('Mail query')
```

Comment:
Actually, if you want to hide `mail_query` field you can specify `hide_blank` option:
`field :mail_query, _('Mail query'), Fields::Field, hide_blank: true`. This will hide the field if value for it is `nil`. Although it doesn't work for table output. So, it's up to you whether have it or not.

Reply:
I don't see a reason to add this if it does not work for table output. Do you?

Reply:
Well, it works for base, csv and other outputs, which are used more for automation purposes. I don't see much usage of this option in the project anyway, so we can leave it as it is now.

Reply:
Sounds good to me :)